### ISSUE TITLE
Scan report: Duplicate issue creation when `artifactName` passed

### ISSUE THREAD

**Issue by tony** [zaproxy/actions-common#20]
![image](https://user-images.githubusercontent.com/26336/144674285-592e7407-cc05-4512-9bde-fdd2019f3f29.png)

Since `readPreviousReport` isn't checking for the artifact name, it will create a new issue when `artifact_name` is passed in through a PR like https://github.com/zaproxy/action-baseline/pull/64

**Comment by tony** [comment]
Closing this as I don't recall the original issue as too much time has passed.  If anyone bumps into this in a future feel free to reopen and cite this.  Thank you!

**Comment by thc202** [comment]
@dbartholomae do you want to take a look at this one? There are still places where `zap_scan` is used but should use the `artifactName` instead. I guess we should also use `artifactName` for the upload without issue creation too.

**Comment by dbartholomae** [comment]
I can look into this, but would like to first make sure that we have a full release cycle running including the GitHub actions, as any change that we currently do otherwise might not yet actually reach any users :)

**Comment by thc202** [comment]
But this introduces a bug, that's why it would be better to address it before the release. (Well, we could always not expose the artifact name in the actions yet.)

**Comment by dbartholomae** [comment]
Oh, I thought that this was a bug that is already in the released code. In that case, I'll look into it before releasing.

**Comment by dbartholomae** [comment]
Are you sure that this introduces a bug? I don't see how the code currently in production is different here from the code that we would release.

**Comment by dbartholomae** [comment]
@tony I've started last year to modernize the package by adding TypeScript, tests, and linting, with the idea to add some features like grouping alert messages by severity in the issues that are created. The next step from my point of view would be to release the code currently in this repo to be used in the actions so that we have a simple release flow. The reason why I commented in this issue is that if I understood @thc202 correctly, they believe that this is a bug that needs to be fixed before we can use the code in this repo for GitHub actions as it would otherwise add a new bug.
My current understanding is that this is an existing bug and should not block releasing the current code.

**Comment by thc202** [comment]
Raised a PR (#60), please take a look.

**Comment by thc202** [comment]
https://github.com/zaproxy/actions-common/issues/20#issuecomment-1493368546

Yes, the custom artifact name was not yet released. As mentioned before (I think), it's true that if we don't use the new feature the bug would not be a problem but IMO better release without the issue in the first place.
